metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/profiler.py

```diff
@@ -240,14 +240,15 @@ def get_bigquery_profile_request(
         dataset_name = BigqueryTableIdentifier(
             project_id=project, dataset=dataset, table=table.name
         ).get_table_name()
-        if not self.is_dataset_eligible_for_profiling(
+        if not profile_table_level_only and not self.is_dataset_eligible_for_profiling(
```

Comment:
some comments might be useful here to make it easier to understand

Reply:
Idea here was that since all this check does is set `profile_table_level_only` to true (and add stuff to the report), there's no point to do the check if `profile_table_level_only` is already true. I don't think it really changes logic, just hides some reporting that didn't actually mean anything. I can add some comments at least though

Reply:
Oh I see - this was purely to suppress the extraneous reporting?

In that case, it should be fine though some comments wouldn't hurt

Reply:
Ugh, it's slightly more complicated because we do some stateful ingestion stuff in `is_dataset_eligible_for_profiling`, will refactor

Reply:
I can take care of this, as I work on sampling in bigquery profiling.

Here is what I did in snowflake for handling this - https://github.com/datahub-project/datahub/blob/master/metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_profiler.py#L127